feat(gnoweb): implement dark theme

mous1985 opened the pull request:
This PR is to implement the dark theme at gnoweb

The color palette for the dark mode was chosen based on the dark mode logo and by inverting the light mode colors.
Here's a preview of what it looks like :

https://github.com/user-attachments/assets/1e91236f-23a5-4bc7-bc56-3f5466b8a3e5



cc @alexiscolin

Gno2D2 commented:
#### 🛠 PR Checks Summary
 🔴 Pending initial approval by a review team member, or review from tech-staff


##### Manual Checks (for Reviewers):
- [ ] **IGNORE** the bot requirements for this PR (force green CI check)
- [ ] The pull request description provides enough details


<details><summary>Read More</summary>

🤖 This bot helps streamline PR reviews by verifying automated checks and providing guidance for contributors and reviewers.

##### ✅ Automated Checks (for Contributors):
 🟢 Maintainers must be able to edit this pull request ([more info](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork))
 🔴 Pending initial approval by a review team member, or review from tech-staff


##### ☑️ Contributor Actions:
1. Fix any issues flagged by automated checks.
2. Follow the Contributor Checklist to ensure your PR is ready for review.
	- Add new tests, or document why they are unnecessary.
	- Provide clear examples/screenshots, if necessary.
	- Update documentation, if required.
	- Ensure no breaking changes, or include `BREAKING CHANGE` notes.
	- Link related issues/PRs, where applicable.

##### ☑️ Reviewer Actions:
1. Complete manual checks for the PR, including the guidelines and additional checks if applicable.

##### 📚 Resources:
- [Report a bug with the bot](https://www.github.com/gnolang/gno/issues/3238).
- [View the bot’s configuration file](https://github.com/gnolang/gno/tree/master/contribs/github-bot/internal/config/config.go).

<details><summary><b>Debug</b></summary><blockquote>
<details><summary><b>Automated Checks</b></summary><blockquote>

<details><summary>Maintainers must be able to edit this pull request (more info)</summary><blockquote>

### If
```
🟢 Condition met
└── 🟢 And
    ├── 🟢 The base branch matches this pattern: ^master$
    └── 🟢 The pull request was created from a fork (head branch repo: mous1985/gno)

```
### Then
```
🟢 Requirement satisfied
└── 🟢 Maintainer can modify this pull request

```
</blockquote></details>

<details><summary>Pending initial approval by a review team member, or review from tech-staff</summary><blockquote>

### If
```
🟢 Condition met
└── 🟢 And
    ├── 🟢 The base branch matches this pattern: ^master$
    └── 🟢 Not (🔴 Pull request author is a member of the team: tech-staff)

```
### Then
```
🔴 Requirement not satisfied
└── 🔴 If
    ├── 🔴 Condition
    │   └── 🔴 Or
    │       ├── 🔴 At least 1 user(s) of the organization reviewed the pull request (with state "APPROVED")
    │       ├── 🔴 At least 1 user(s) of the team tech-staff reviewed pull request
    │       └── 🔴 This pull request is a draft
    └── 🔴 Else
        └── 🔴 And
            ├── 🟢 This label is applied to pull request: review/triage-pending
            └── 🔴 On no pull request

```
</blockquote></details>

</blockquote></details>


<details><summary><b>Manual Checks</b></summary><blockquote>

<details><summary>**IGNORE** the bot requirements for this PR (force green CI check)</summary><blockquote>

### If
```
🟢 Condition met
└── 🟢 On every pull request

```
### Can be checked by

- Any user with comment edit permission

</blockquote></details>

<details><summary>The pull request description provides enough details</summary><blockquote>

### If
```
🟢 Condition met
└── 🟢 And
    ├── 🟢 Not (🔴 Pull request author is a member of the team: core-contributors)
    └── 🟢 Not (🔴 Pull request author is user: dependabot[bot])

```
### Can be checked by
 - team core-contributors

</blockquote></details>

</blockquote></details>

</blockquote></details>

</details>

alexiscolin commented:
Hey @mous1985 Thank you for starting this PR, I was waiting for someone to init it in order to take care of the dark theme. There’s quite a bit to review—both on the code and design side—and this is an important change, so I’ll be giving it a close look. Please feel free to ping me whenever you’re ready for more eyes on it. I’m happy to dive in and help out, especially since it’s such a core part of the design.

mous1985 commented:
Hi @alexiscolin , thanks for your feedback, I thought I'd start with the configuration and I'd work with you on the UI and design.
I'll fix some issues in CI and i'll keep you informed :) .

alexiscolin commented on gno.land/pkg/gnoweb/components/layouts/head.html:
```diff
@@ -44,5 +44,11 @@
+  <meta name="color-scheme" content="light dark" />
+
+  <!-- mobile Theme Color -->
+  <meta name="theme-color" light="#FFFFFF" dark="dark" />
```
`light="#FFFFFF" dark="dark"` doesn't exist in the meta `theme-color` and has no effect on the browser UI color as expected. 

```suggestion
<meta name="theme-color" content="#FFFFFF" media="(prefers-color-scheme: light)" />
<meta name="theme-color" content="#000000" media="(prefers-color-scheme: dark)" />
```

alexiscolin commented on gno.land/pkg/gnoweb/frontend/js/theme.ts:
```diff
@@ -0,0 +1,82 @@
+class Theme {
```
Remove the JS please and let's use `prefers-color-scheme` instead as we want to avoid JS when possible.

alexiscolin commented on gno.land/pkg/gnoweb/frontend/css/tx.config.js:
```diff
@@ -39,10 +41,22 @@ export default {
       green: {
         400: "#2D8D72", // Primary dark color
         600: "#226C57", // Primary light color
+        // dark mode  
```
This is not a green but a white. This colors already exists in the palette: `light`.

alexiscolin replied:
I can handle the palette updates myself if you grant me access to your repository. It might be quicker and easier than multiple back-and-forth iterations. 👍

alexiscolin commented on gno.land/pkg/gnoweb/frontend/css/tx.config.js:
```diff
@@ -39,10 +41,22 @@ export default {
         800: "#131313", // Background dark color
         900: "#080809", // Title color
+        // dark mode grays
+        d50: "#121212",    // Softer than pure black for backgrounds
```
Because we use the palette directly in Tailwind (without separate utility tokens), it’s best to keep things consistent. We can merge our light and dark colors into a single, graded scale (e.g., 50 to 900) so that the dark: variants simply invert the shades. If a palette gets too crowded or if different colors serve distinct purposes, we can split them by family (like grays and greens) to maintain clarity and ease of use.

alexiscolin replied:
I can handle the palette updates myself if you grant me access to your repository. It might be quicker and easier than multiple back-and-forth iterations. 👍

alexiscolin commented on gno.land/pkg/gnoweb/components/layouts/index.html:
```diff
@@ -1,7 +1,7 @@
 
 <!DOCTYPE html>
-<html lang="en">
+<html lang="en" class="dark">
```
We should't avoid relying on JS `class selector`

alexiscolin commented on gno.land/pkg/gnoweb/frontend/css/input.css:
```diff
@@ -39,16 +39,33 @@
+  }
+
+  html.dark {
+    @apply bg-dark text-gray-d600;
```
Should be used within regular `html` element.

alexiscolin commented on gno.land/pkg/gnoweb/components/ui/logo.html:
```diff
@@ -4,42 +4,52 @@
     <path
       d="M15.7782 18.1953C15.5729 17.4557 15.1344 16.8048 14.5394 16.3075C14.3097 16.1152 14.0591 15.9332 13.7854 15.7636C13.5534 15.6191 13.261 15.8319 13.3283 16.0936L13.493 16.7331C13.7018 17.5456 12.8133 18.2044 12.0732 17.7868L10.0838 16.6648C8.77648 15.9275 7.16872 15.9275 5.8614 16.6648L3.872 17.7868C3.13192 18.2044 2.24336 17.5444 2.45216 16.7331L2.62152 16.0754C2.6888 15.8148 2.39764 15. ...
       fill="#999999"
+      class="dark:[fill:#949494]"
```
Why not using `fill` utility and merging the `fill` attribute above in one regular `class` attr instead?

alexiscolin commented on gno.land/pkg/gnoweb/components/ui/icons.html:
```diff
@@ -217,5 +217,25 @@
       />
     </symbol>
-  </svg>
+    <symbol id="ico-moon" viewBox="0 0 24 24">
```
Where does those icons come from? What licence? 
Anyway, we don't need icons without JS.

alexiscolin reviewed (changes requested):
@mous1985, thank you for initiating this much-needed work—it's a great start. I noticed some issues with how the palettes are used and how the color system is managed. I've started reviewing the major points, but there are also a lot of small UI/UX optimizations needed in the color scheme.

**Would you mind giving me access to your repository so I can make the necessary fixes**? It would be much easier than updating each value individually while adjusting the entire palette at the same time (I started the review and... yeah it's harder to manage all these very small stuffs in a review). If you're okay with that, I'd be happy to continue building on your nice work directly in your repo. 👍

mous1985 commented:
> @mous1985, thank you for initiating this much-needed work—it's a great start. I noticed some issues with how the palettes are used and how the color system is managed. I've started reviewing the major points, but there are also a lot of small UI/UX optimizations needed in the color scheme.
> 
> **Would you mind giving me access to your repository so I can make the necessary fixes**? It would be much easier than updating each value individually while adjusting the entire palette at the same time (I started the review and... yeah it's harder to manage all these very small stuffs in a review). If you're okay with that, I'd be happy to continue building on your nice work directly in your repo. 👍

Hi @alexiscolin , thank you for your feedback, i added you in my repo, and i'm also looking at the comments  and i'm resolving them

mous1985 commented:
Hi @alexiscolin 
Sorry for the delay 🙏🏽  

- [ ]  I removed  `theme.ts` and use `prefers-color-scheme` [4d8031b](https://github.com/gnolang/gno/pull/3869/commits/4d8031bc7d12cc52e4c4725d74a027d028980ede)

- [ ] I found the Icons sun and moon  on github it's open source but i removed it [635bb55](https://github.com/gnolang/gno/pull/3869/commits/635bb55dc77362552a13c51a6bc600d848e669f8) 

- [ ]  For the logo I merge the `fill`  into a single `class`  [2df6a48](https://github.com/gnolang/gno/pull/3869/commits/2df6a4814af48a690c89539bbd2ff0f796702f00)

- [ ]  I've add you in the repo for the color palette, i'm waiting for your suggestions :)

thehowl commented:
Closing as stale, @alexiscolin to take care of making a new PR I believe.

[dark reader](https://darkreader.org/) works well for those who need a solution in the meantime.
